### PR TITLE
neomutt: build against homebrew libiconv

### PR DESCRIPTION
**Pull request by roccoblues**
libiconv shipped with the latest macOS version is broken/incompatible with Neomutt. It leads to improper handling of utf8 or umlauts. Upstream bug report: https://github.com/neomutt/neomutt/issues/4084

This updates neomutt to the latest version and fixes the iconv dependency.

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/n/neomutt.rb [hunk 1]
```diff
@@ -58,6 +59,7 @@ def install
       --with-ncurses=#{Formula["ncurses"].opt_prefix}
       --with-ssl=#{Formula["openssl@3"].opt_prefix}
       --with-sqlite=#{Formula["sqlite"].opt_prefix}
+      EXTRA_CFLAGS=-I#{Formula["libiconv"].opt_prefix}/include
```
Comment (SMillerDev) [inline Formula/n/neomutt.rb:62]: Can it not be configured the usual way?

Comment (roccoblues) [reply Formula/n/neomutt.rb:61]: Unfortunately there's no `with-libiconv=` currently.

Comment (SMillerDev) [reply Formula/n/neomutt.rb:61]: Could you make a pull request upstream to add that then, we need to work with upstream to fix this.

Comment (flatcap) [reply Formula/n/neomutt.rb:61]: I've hacked something together: https://github.com/neomutt/neomutt/pull/4085

I need to get second opinion on the change.
I'm planning another bug-fix release late next week.

**Comment by Bo98** [comment]
So is the issue you are seeing that the new system iconv is behaving as if `TRANSLIT` is always enabled? Sounds like the issue about I've talked to Apple about: "iconv in Sonoma behaves as TRANSLIT is always enabled but incorrectly returns 0" (FB13166630).

Might be worth mentioning that as a comment in the formula as it's possible this will be fixed in the future.

**Comment by roccoblues** [comment]
> So is the issue you are seeing that the new BSD-like system iconv is behaving as if `TRANSLIT` is always enabled?

tbh, I'm not deep enough into the topic to answer that. I only noticed that switching to the Homebrew iconv version fixed the umlaut in my last name. 😏 

> Sounds like the issue I've talked to Apple about: "iconv in Sonoma behaves as TRANSLIT is always enabled but incorrectly returns 0" (FB13166630).

Is that issue somewhere publicly available? I tried to google but couldn't find anything.

File: Formula/n/neomutt.rb [hunk 2]
```diff
@@ -22,6 +22,7 @@ class Neomutt < Formula
   depends_on "tcl-tk" => :build
   depends_on "gettext"
   depends_on "gpgme"
+  depends_on "libiconv"
```
Comment (carlocab) [inline Formula/n/neomutt.rb:25]: This needs to be macOS-only.

Comment (flatcap) [reply Formula/n/neomutt.rb:25]: How about?

```ruby
on_macos do
  # macOS Sonoma ships with a broken libiconv
  depends_on "libiconv"
end
``` 

@roccoblues 
Unfortunately, you made your own fork, so I can update the PR.

Comment (roccoblues) [reply Formula/n/neomutt.rb:25]: I think we also need to guard the configure option. My plan is to update this PR when the new version is ready.

**Comment by flatcap** [comment]
[NeoMutt 2023-11-03](https://github.com/neomutt/neomutt/releases/tag/20231103) has been released.
It includes fixes that should help:
- https://github.com/neomutt/neomutt/commit/52e6ea650426ed7334fc037e91df6cf093d76f3b Add `./configure --with-iconv=PATH`
- https://github.com/neomutt/neomutt/commit/50da69721db8cea66ab06c1fba0650b62763f517 Fix configure path for ncurses

**Comment by roccoblues** [comment]
@SMillerDev @carlocab please check this out again when you have time. I think I've addressed your comments.

Only the "Commit style" check fails. I don't really understand why, maybe because it's looking at single commits and not the PR title?

**Comment by fxcoudert** [comment]
Rebased, squashed the commits, added a comment. Should be suitable now.

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
@roccoblues thanks for your first contribution to homebrew-core! 🎉

**Pull request merged**